juju/model.py

```diff
@@ -86,22 +118,38 @@ def __getitem__(self, key):
         return self._data[key]
 
 
-class RelationMapping(LazyMapping):
+class RelationMapping(Mapping):
     """Map of relation names to lists of Relation instances."""
     def __init__(self, relation_names, local_unit, backend, cache):
-        self._relation_names = relation_names
+        self._data = {}
+        for relation_name in relation_names:
+            self._data[relation_name] = RelationList(relation_name, local_unit, backend, cache)
```

Comment:
Instead of all the extra complexity wouldn't it be easier to simply not load the relation list until people actually ask for the list, but then really load that list instead of returning the object that looks like it but loads when touched?  The complete loading of the dictionary at once is what causes the complete listing.